clients/http_json_rpc_client.go

```diff
@@ -524,14 +557,11 @@ func getJsonRpcResponseFromNode(rootNode ast.Node) (*common.JsonRpcResponse, err
 		}
 
 		cause := fmt.Sprintf("cannot parse json rpc response from upstream, for result: %s, for error: %s", rawResult, rawError)
-		jrResp = &common.JsonRpcResponse{
-			Result: util.Str2Mem(rawResult),
-			Error: common.NewErrJsonRpcExceptionExternal(
-				int(common.JsonRpcErrorParseException),
-				cause,
-				"",
-			),
-		}
+		jrResp.Error = common.NewErrJsonRpcExceptionExternal(
+			int(common.JsonRpcErrorParseException),
+			cause,
+			"",
+		)
```

Comment:
Probably it's good to somehow pass the rawResult in the error, maybe via "Details" in case users need to investigate what was the broken result?

Reply:
Isn't it already passed in `cause`?